server/test/unit/server/webservices/views/test_search.py

```diff
@@ -266,6 +267,17 @@ class FakeSearchView(search.SearchView):
         self.assertEqual([c[1][0] for c in FakeSearchView.serializer.mock_calls],
                          ['big money', 'bigger money'])
 
+    def test__generate_response_with_invalid_criteria(self):
+        """
+        Test that a pymongo exception is handled correctly.
+        """
+        class FakeSearchView(search.SearchView):
+            model = mock.MagicMock()
+
+        query = {'filters': {'money': {'$gt': 1000000}}}
```

Comment:
![image](https://cloud.githubusercontent.com/assets/441252/10677569/3eeebaaa-78db-11e5-91ff-46e173c97e83.png)

It lets you drag and drop images, which I think is good enough...